src/components/views/rooms/RoomKnocksBar.tsx

```diff
@@ -0,0 +1,152 @@
+import { Icon as XIcon } from "../../../../res/img/feather-customised/x.svg";
+import dis from "../../../dispatcher/dispatcher";
+import { useTypedEventEmitterState } from "../../../hooks/useEventEmitter";
+import { _t } from "../../../languageHandler";
+import Modal from "../../../Modal";
+import MemberAvatar from "../avatars/MemberAvatar";
+import ErrorDialog from "../dialogs/ErrorDialog";
+import { RoomSettingsTab } from "../dialogs/RoomSettingsDialog";
+import AccessibleButton from "../elements/AccessibleButton";
+import Heading from "../typography/Heading";
+
+export const RoomKnocksBar: VFC<{ room: Room }> = ({ room }) => {
+    const client = room.client;
+    const userId = client.getUserId() || "";
+    const canInvite = room.canInvite(userId);
+    const member = room.getMember(userId);
+    const state = room.getLiveTimeline().getState(EventTimeline.FORWARDS);
+    const canKick = member && state ? state.hasSufficientPowerLevelFor("kick", member.powerLevel) : false;
+
+    const handleApprove = (userId: string): void => {
+        setDisabled(true);
+        client.invite(room.roomId, userId).catch(onError);
+    };
+
+    const handleDeny = (userId: string): void => {
+        setDisabled(true);
+        client.kick(room.roomId, userId).catch(onError);
+    };
+
+    const handleOpenRoomSettings = (): void =>
+        dis.dispatch({ action: "open_room_settings", room_id: room.roomId, initial_tab_id: RoomSettingsTab.People });
+
+    const onError = (error: MatrixError): void => {
+        setDisabled(false);
+        Modal.createDialog(ErrorDialog, { title: error.name, description: error.message });
+    };
+
+    const [disabled, setDisabled] = useState(false);
+    const knockMembers = useTypedEventEmitterState(
+        room,
+        RoomStateEvent.Members,
+        useCallback(() => room.getMembersWithMembership("knock"), [room]),
+    );
+    const knockMembersCount = knockMembers.length;
+
+    if (room.getJoinRule() !== JoinRule.Knock || knockMembersCount === 0 || (!canInvite && !canKick)) return null;
+
+    let buttons: ReactElement = (
+        <AccessibleButton
+            className="mx_RoomKnocksBar_action"
+            kind="primary"
+            onClick={handleOpenRoomSettings}
+            title={_t("action|view")}
+        >
+            {_t("action|view")}
+        </AccessibleButton>
+    );
+    let names: string = knockMembers
+        .slice(0, 2)
+        .map((knockMember) => knockMember.name)
+        .join(", ");
+    let link: ReactNode = null;
+    switch (knockMembersCount) {
+        case 1: {
+            buttons = (
+                <>
+                    <AccessibleButton
+                        className="mx_RoomKnocksBar_action"
+                        disabled={!canKick || disabled}
+                        kind="icon_primary_outline"
+                        onClick={() => handleDeny(knockMembers[0].userId)}
+                        title={_t("Deny")}
```

Comment:
What is the state of translations at the moment? Are some in the old form like this and some in the new e.g. `action|view`? Maybe it's all in the new form and these just slipped through?

Reply:
Nice catch! Will change it to `action|…`